### PR TITLE
Update IAM CRD to allow remediationAction enforce

### PR DESCRIPTION
**Pull request by JustinKuli**
Refs:
 - https://github.com/stolostron/backlog/issues/22322

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

### CODE REVIEW

File: pkg/addon/iampolicy/manifests/managedclusterchart/templates/policy.open-cluster-management.io_iampolicy_crd.yaml [hunk 1]
```diff
@@ -82,6 +82,8 @@ spec:
               enum:
               - Inform
               - inform
+              - Enforce
```
Comment (mprahl) [inline pkg/addon/iampolicy/manifests/managedclusterchart/templates/policy.open-cluster-management.io_iampolicy_crd.yaml:85]: Could you please also update the description of the property?

Comment (JustinKuli) [reply pkg/addon/iampolicy/manifests/managedclusterchart/templates/policy.open-cluster-management.io_iampolicy_crd.yaml:85]: You mean the `description: enforce, inform`?

I wouldn't change it here, but we could update it in the iam-policy repo: https://github.com/stolostron/iam-policy-controller/blob/main/api/v1/iampolicy_types.go#L56

Comment (mprahl) [reply pkg/addon/iampolicy/manifests/managedclusterchart/templates/policy.open-cluster-management.io_iampolicy_crd.yaml:85]: You're right, I had accidentally changed the description in the wrong spot:
https://github.com/stolostron/iam-policy-controller/pull/122

**Pull request closed without merging**